<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="twitter:card" content="summary_large_image" /><!-- Begin Jekyll SEO tag v2.6.1 -->
<title>Title | Jasmine G. Young - Data Scientist</title>
<meta name="generator" content="Jekyll v4.1.1" />
<meta property="og:title" content="Title" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="This blog contains various data science projects in Jasmine’s portfolio." />
<meta property="og:description" content="This blog contains various data science projects in Jasmine’s portfolio." />
<link rel="canonical" href="https://jgy4.github.io/fp.github/2021/10/16/Test.html" />
<meta property="og:url" content="https://jgy4.github.io/fp.github/2021/10/16/Test.html" />
<meta property="og:site_name" content="Jasmine G. Young - Data Scientist" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2021-10-16T00:00:00-05:00" />
<script type="application/ld+json">
{"datePublished":"2021-10-16T00:00:00-05:00","url":"https://jgy4.github.io/fp.github/2021/10/16/Test.html","@type":"BlogPosting","headline":"Title","dateModified":"2021-10-16T00:00:00-05:00","mainEntityOfPage":{"@type":"WebPage","@id":"https://jgy4.github.io/fp.github/2021/10/16/Test.html"},"description":"This blog contains various data science projects in Jasmine’s portfolio.","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/fp.github/assets/css/style.css"><link type="application/atom+xml" rel="alternate" href="https://jgy4.github.io/fp.github/feed.xml" title="Jasmine G. Young - Data Scientist" /><link rel="shortcut icon" type="image/x-icon" href="/fp.github/images/favicon.ico"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/Primer/15.2.0/primer.css" integrity="sha512-xTz2ys4coGAOz8vuV1NcQBkgVmKhsSEtjbqyMJbBHRplFuvKIUo6xhLHpAyPt9mfR6twHJgn9OgVLuqOvjeBhg==" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css" integrity="sha512-1PKOgIY59xJ8Co8+NE6FZ+LOAZKjy+KY8iq0G4B3CyeY6wYHN3yt9PW0XpSriVlkMXe40PTKnXrLnZ9+fkDaog==" crossorigin="anonymous" />


    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css" integrity="sha512-h7nl+xz8wgDlNM4NqKEM4F1NkIRS17M9+uJwIGwuo8vGqIl4BhuCKdxjWEINm+xyrUjNCnK5dCrhM0sj+wTIXw==" crossorigin="anonymous" />
    <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.js" integrity="sha512-/CMIhXiDA3m2c9kzRyd97MTb3MC6OVnx4TElQ7fkkoRghwDf6gi41gaT1PwF270W6+J60uTmwgeRpNpJdRV6sg==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/contrib/auto-render.min.js" integrity="sha512-Do7uJAaHZm5OLrIv/yN4w0iG1dbu01kzdMNnFfu/mAqgUk6Nniv2JYHcwH+cNwjqgLcqcuBBk+JRvprLVI8azg==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A==" crossorigin="anonymous"></script>
    <script>
    document.addEventListener("DOMContentLoaded", function() {
        renderMathInElement( document.body, {
        delimiters: [
            {left: "$$", right: "$$", display: true},
            {left: "[%", right: "%]", display: true},
            {left: "$", right: "$", display: false}
        ]}
        );
    });
    </script>


<script>
function wrap_img(fn) {
    if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
        var elements = document.querySelectorAll(".post img");
        Array.prototype.forEach.call(elements, function(el, i) {
            if (el.getAttribute("title") && (el.className != "emoji")) {
                const caption = document.createElement('figcaption');
                var node = document.createTextNode(el.getAttribute("title"));
                caption.appendChild(node);
                const wrapper = document.createElement('figure');
                wrapper.className = 'image';
                el.parentNode.insertBefore(wrapper, el);
                el.parentNode.removeChild(el);
                wrapper.appendChild(el);
                wrapper.appendChild(caption);
            }
        });
    } else { document.addEventListener('DOMContentLoaded', fn); }
}
window.onload = wrap_img;
</script>

<script>
    document.addEventListener("DOMContentLoaded", function(){
    // add link icon to anchor tags
    var elem = document.querySelectorAll(".anchor-link")
    elem.forEach(e => (e.innerHTML = '<i class="fas fa-link fa-xs"></i>'));
    });
</script>
</head>
<body><header class="site-header">

  <div class="wrapper"><a class="site-title" rel="author" href="/fp.github/">Jasmine G. Young - Data Scientist</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/fp.github/about/">About Me</a><a class="page-link" href="/fp.github/search/">Search</a><a class="page-link" href="/fp.github/categories/">Tags</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">Title</h1><p class="post-meta post-meta-title"><time class="dt-published" datetime="2021-10-16T00:00:00-05:00" itemprop="datePublished">
        Oct 16, 2021
      </time>
       • <span class="read-time" title="Estimated read time">
    
    
      7 min read
    
</span></p>

    

    
      
        <div class="pb-5 d-flex flex-wrap flex-justify-end">
          <div class="px-2">

    <a href="https://github.com/jgy4/fp.github/tree/master/_notebooks/2021-10-16-Test.ipynb" role="button" target="_blank">
<img class="notebook-badge-image" src="/fp.github/assets/badges/github.svg" alt="View On GitHub">
    </a>
</div>

          <div class="px-2">
    <a href="https://mybinder.org/v2/gh/jgy4/fp.github/master?filepath=_notebooks%2F2021-10-16-Test.ipynb" target="_blank">
        <img class="notebook-badge-image" src="/fp.github/assets/badges/binder.svg" alt="Open In Binder"/>
    </a>
</div>

          <div class="px-2">
    <a href="https://colab.research.google.com/github/jgy4/fp.github/blob/master/_notebooks/2021-10-16-Test.ipynb" target="_blank">
        <img class="notebook-badge-image" src="/fp.github/assets/badges/colab.svg" alt="Open In Colab"/>
    </a>
</div>
        </div>
      </header>

  <div class="post-content e-content" itemprop="articleBody">
    <!--
#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: _notebooks/2021-10-16-Test.ipynb
-->

<div class="container" id="notebook-container">
        
    
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">dash</span>
<span class="kn">from</span> <span class="nn">dash.dependencies</span> <span class="kn">import</span> <span class="n">Input</span><span class="p">,</span> <span class="n">Output</span>
<span class="kn">from</span> <span class="nn">dash</span> <span class="kn">import</span> <span class="n">dcc</span>
<span class="kn">from</span> <span class="nn">dash</span> <span class="kn">import</span> <span class="n">html</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">datetime</span> <span class="kn">import</span> <span class="n">datetime</span> <span class="k">as</span> <span class="n">dt</span>

<span class="n">table2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s1">&#39;tab001.xlsx&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
<span class="n">table2</span> <span class="o">=</span> <span class="n">table2</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;-&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
<span class="n">table2</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">app</span> <span class="o">=</span> <span class="n">dash</span><span class="o">.</span><span class="n">Dash</span><span class="p">(</span><span class="s1">&#39;Hello World&#39;</span><span class="p">)</span>

<span class="n">app</span><span class="o">.</span><span class="n">layout</span> <span class="o">=</span> <span class="n">html</span><span class="o">.</span><span class="n">Div</span><span class="p">([</span>
    <span class="n">dcc</span><span class="o">.</span><span class="n">Dropdown</span><span class="p">(</span>
        <span class="nb">id</span><span class="o">=</span><span class="s1">&#39;my-dropdown&#39;</span><span class="p">,</span>
        <span class="n">options</span><span class="o">=</span><span class="p">[</span>
            <span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;Doctorate Recipients&#39;</span><span class="p">,</span> <span class="s1">&#39;value&#39;</span><span class="p">:</span> <span class="s1">&#39;Doctorate recipients&#39;</span><span class="p">},</span>
            <span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;Percent Change&#39;</span><span class="p">,</span> <span class="s1">&#39;value&#39;</span><span class="p">:</span> <span class="s1">&#39;</span><span class="si">% c</span><span class="s1">hange from previous year&#39;</span><span class="p">}</span>
        <span class="p">],</span>
        <span class="n">value</span><span class="o">=</span><span class="s1">&#39;Doctorate Recipients&#39;</span>
    <span class="p">),</span>
    <span class="n">dcc</span><span class="o">.</span><span class="n">Graph</span><span class="p">(</span><span class="nb">id</span><span class="o">=</span><span class="s1">&#39;my-graph&#39;</span><span class="p">)</span>
<span class="p">],</span> <span class="n">style</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;width&#39;</span><span class="p">:</span> <span class="s1">&#39;500&#39;</span><span class="p">})</span>

<span class="nd">@app</span><span class="o">.</span><span class="n">callback</span><span class="p">(</span><span class="n">Output</span><span class="p">(</span><span class="s1">&#39;my-graph&#39;</span><span class="p">,</span> <span class="s1">&#39;figure&#39;</span><span class="p">),</span> <span class="p">[</span><span class="n">Input</span><span class="p">(</span><span class="s1">&#39;my-dropdown&#39;</span><span class="p">,</span> <span class="s1">&#39;value&#39;</span><span class="p">)])</span>
<span class="k">def</span> <span class="nf">update_graph</span><span class="p">(</span><span class="n">selected_dropdown_value</span><span class="p">):</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">table2</span>
    <span class="k">return</span> <span class="p">{</span>
        <span class="s1">&#39;data&#39;</span><span class="p">:</span> <span class="p">[{</span>
            <span class="s1">&#39;x&#39;</span><span class="p">:</span> <span class="n">df</span><span class="o">.</span><span class="n">Year</span><span class="p">,</span>
            <span class="s1">&#39;y&#39;</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">selected_dropdown_value</span><span class="p">)]</span>
        <span class="p">}],</span>
        <span class="s1">&#39;layout&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;margin&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;l&#39;</span><span class="p">:</span> <span class="mi">40</span><span class="p">,</span> <span class="s1">&#39;r&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;t&#39;</span><span class="p">:</span> <span class="mi">20</span><span class="p">,</span> <span class="s1">&#39;b&#39;</span><span class="p">:</span> <span class="mi">30</span><span class="p">}}</span>
    <span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
    

    
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">app</span><span class="o">.</span><span class="n">run_server</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stdout output_text">
<pre>Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

Dash is running on http://127.0.0.1:8050/

 * Serving Flask app &#34;Hello World&#34; (lazy loading)
 * Environment: production
<span class="ansi-red-fg">   WARNING: This is a development server. Do not use it in a production deployment.</span>
   Use a production WSGI server instead.
 * Debug mode: off
</pre>
</div>
</div>

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre> * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)
127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET / HTTP/1.1</span>&#34; 200 -
127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET /_dash-layout HTTP/1.1</span>&#34; 200 -
127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET /_dash-dependencies HTTP/1.1</span>&#34; 200 -
127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET /_dash-component-suites/dash/dcc/async-dropdown.js HTTP/1.1</span>&#34; 200 -
127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET /_dash-component-suites/dash/dcc/async-graph.js HTTP/1.1</span>&#34; 200 -
</pre>
</div>
</div>

<div class="output_area">

<div class="output_subarea output_stream output_stdout output_text">
<pre>Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
</pre>
</div>
</div>

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-white-fg">GET /_dash-component-suites/dash/dcc/async-plotlyjs.js HTTP/1.1</span>&#34; 200 -
</pre>
</div>
</div>

<div class="output_area">

<div class="output_subarea output_stream output_stdout output_text">
<pre>Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File &#34;pandas/_libs/index.pyx&#34;, line 70, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/index.pyx&#34;, line 97, in pandas._libs.index.IndexEngine.get_loc
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File &#34;pandas/_libs/hashtable_class_helper.pxi&#34;, line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: &#39;Doctorate Recipients&#39;

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/dash.py&#34;, line 1336, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/dash/_callback.py&#34;, line 151, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File &#34;&lt;ipython-input-19-2ba5f7baee40&gt;&#34;, line 32, in update_graph
    &#39;y&#39;: df[str(selected_dropdown_value)]
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py&#34;, line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File &#34;/Users/jasmineyoung/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py&#34;, line 2891, in get_loc
    raise KeyError(key) from err
KeyError: &#39;Doctorate Recipients&#39;
</pre>
</div>
</div>

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>127.0.0.1 - - [16/Oct/2021 23:48:18] &#34;<span class="ansi-magenta-intense-fg ansi-bold">POST /_dash-update-component HTTP/1.1</span>&#34; 500 -
127.0.0.1 - - [16/Oct/2021 23:48:21] &#34;<span class="ansi-white-fg">POST /_dash-update-component HTTP/1.1</span>&#34; 200 -
127.0.0.1 - - [16/Oct/2021 23:48:24] &#34;<span class="ansi-white-fg">POST /_dash-update-component HTTP/1.1</span>&#34; 200 -
</pre>
</div>
</div>

</div>
</div>

</div>
    

</div>



  </div><a class="u-url" href="/fp.github/2021/10/16/Test.html" hidden></a>
</article>
      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/fp.github/"></data>

  <div class="wrapper">

    <div class="footer-col-wrapper">
      <div class="footer-col">
        <p class="feed-subscribe">
          <a href="/fp.github/feed.xml">
            <svg class="svg-icon orange">
              <use xlink:href="/fp.github/assets/minima-social-icons.svg#rss"></use>
            </svg><span>Subscribe</span>
          </a>
        </p>
      </div>
      <div class="footer-col">
        <p>This blog contains various data science projects in Jasmine&#39;s portfolio.</p>
      </div>
    </div>

    <div class="social-links"><ul class="social-media-list"><li><a rel="me" href="https://github.com/https%3A%2F%2Fgithub.com%2Fjgy4" title="https://github.com/jgy4"><svg class="svg-icon grey"><use xlink:href="/fp.github/assets/minima-social-icons.svg#github"></use></svg></a></li></ul>
</div>

  </div>

</footer>
</body>

</html>
